sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -933,10 +933,14 @@ def exists(self, timeout=None, version_id=None):
         :returns: boolean
         """
         try:
-            self._client.blob.get_properties(
-                timeout=timeout,
-                version_id=version_id)
-            return True
+            blob_props = self._client.blob.get_properties(
+                timeout=kwargs.pop('timeout', None),
+                version_id=kwargs.pop('version_id', None),
+                snapshot=self.snapshot,
+                cls=deserialize_blob_properties)
+            if blob_props and blob_props.is_current_version or blob_props and self.snapshot:
```

Comment:
no need to set cls?
we don't need to have the if check right?

Reply:
We do need to set cls.
As for the if check we need it because older blob versions that are overwritten still have properties. That is why we need to check for current version. Snapshots are not current versions that is why we also nee to check if it is a snapshot blob.

Reply:
It doesn't matter whether the older blob versions/snapshot have properties or not if I understand properly.
when you call get properties, you are only checking what you want to check. 
If you want to check if a version exists, you want to pass in version_id, else it's checking the root blob. If you want to check if the blob snapshot exists, you want to set snapshot on when creating blob_client. You can not check the version, snapshot and root blob within just a single call.